subprojects/core/src/main/java/org/gradle/configuration/project/LifecycleProjectEvaluator.java

```diff
@@ -76,7 +77,13 @@ private void notifyAfterEvaluate(ProjectEvaluationListener listener, ProjectInte
         } catch (Exception e) {
             if (state.hasFailure()) {
                 // Just log this failure, and pass the existing failure out in the project state
-                LOGGER.error("Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.", e);
+                boolean logStackTraces = project.getGradle().getStartParameter().getShowStacktrace() != ShowStacktrace.INTERNAL_EXCEPTIONS;
+                String infoMessage = "Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.";
```

Comment:
I think it's a clearer representation of what's going on for users. I'd vote to go with that.